pkg/controllers/deprovisioning/controller.go

```diff
@@ -115,32 +116,13 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 	// Attempt different deprovisioning methods. We'll only let one method perform an action
 	isConsolidated := c.cluster.Consolidated()
 	for _, d := range c.deprovisioners {
-		candidates, err := GetCandidates(ctx, c.cluster, c.kubeClient, c.clock, c.cloudProvider, d.ShouldDeprovision)
+		success, err := c.deprovision(ctx, d)
```

Comment:
I think we need more logging until we are 100% sure that issues around deprovisioning are resolved.

e.g.
- Are we hung in cluster synced?
- Is the deprovisioning loop runnning?
- What deprovisioners are running?

Reply:
I'd like to do this in another PR.